### ISSUE TITLE
[BETA] Mir-algorithm compiler segmentation failure 

### ISSUE THREAD

**Issue by 9il** [ldc-developers/ldc#2729]
Hi
LDC beta fails in MirAlgorithm's Travis https://travis-ci.org/libmir/mir-algorithm/jobs/389055282

**Comment by kinke** [comment]
Thx for the report. Reduced, failing with `ldc2 -c -mtriple=x86_64-linux-gnu`:
```D
struct ExtMulResult
{
    ulong low;
    ulong high;
}

ExtMulResult extMul(in ulong a, in ulong b) @nogc nothrow pure @safe
{
    import ldc.simd;
    auto r = inlineIR!(`
        %a = zext i64 %0 to i128
        %b = zext i64 %1 to i128
        %m = mul i128 %a, %b
        %n = lshr i128 %m, 64
        %h = trunc i128 %n to i64
        %l = trunc i128 %m to i64
        %agg1 = insertvalue [2 x i64] undef, i64 %l, 0
        %agg2 = insertvalue [2 x i64] %agg1, i64 %h, 1
        ret [2 x i64] %agg2`, ulong[2])(a, b);

    return ExtMulResult(r[0], r[1]);
}
```

**Comment by kinke** [comment]
This apparently regressed (v1.9 works) because of a 2.080 Posix x86_64 ABI fix - `ulong[2]` is now returned in registers, not via sret.

**Comment by kinke** [comment]
[Btw, using `return inlineIR!(..., ExtMulResult);` directly would be nicer.]

**Comment by kinke** [comment]
Fixed by #2732.
